tests/unit/htdocs/modules/system/FineUploaderPathTest.php

```
@@ -0,0 +1,147 @@
+/**
+ * Path-confinement guards for the FineUploader handler (findings H-8 / A2-H-2).
+ *
+ * SystemFineUploadHandler is abstract, so we exercise its protected confinement
+ * helpers through a concrete subclass and reflection, and assert the image/avatar
+ * subclasses inherit (do not override) the confined methods.
+ */
+final class FineUploaderPathTest extends TestCase
+{
+    private function handler(): \SystemFineUploadHandler
+    {
+        return new class (new \stdClass()) extends \SystemFineUploadHandler {
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '24', column '31'). (undefined)

(MissingImport)